### ISSUE TITLE
In-Game: [macOS] buffer_save() does not reliably create the expected files

### ISSUE THREAD

**Issue by NanoSharkGames** [YoYoGames/GameMaker-Bugs#2399]
### Description
Files aren't being saved when using a buffer & the buffer_save function. Debug output also shows that working_directory does not match where files would be saved.

### Steps To Reproduce
1. Run project.
2. Verify that "test.txt" file is not in working_directory.

[5eb2fefe-91d4-4aea-a4cf-1bda3217486d](https://api.gamemaker.io/api/github/downloads/5eb2fefe-91d4-4aea-a4cf-1bda3217486d)

**Comment by nuevocharrua** [comment]
Hi,

I wanted to add that the files saved using buffer_save should not necessarily be in the working_directory. Instead, they should be in the game_save_id directory. However, I've also noticed that buffer_save and file_exists are not functioning correctly in the game_save_id directory either.

It's worth mentioning that I've only been able to reproduce this issue on macOS. On Windows, these functions seem to be working fine in my experience.

Just thought this additional information might help in pinpointing the issue.

Thanks!

**Comment by NanoSharkGames** [comment]
> Hi,
> 
> I wanted to add that the files saved using buffer_save should not necessarily be in the working_directory. Instead, they should be in the game_save_id directory. However, I've also noticed that buffer_save and file_exists are not functioning correctly in the game_save_id directory either.
> 
> It's worth mentioning that I've only been able to reproduce this issue on macOS. On Windows, these functions seem to be working fine in my experience.
> 
> Just thought this additional information might help in pinpointing the issue.
> 
> Thanks!

You're right about where the files are saved (in game_save_id), my mistake. Also, I did not initially mention that this is happening on macOS, not Windows. I updated the bug report title to reflect that.

Thanks for bringing that to my attention!

**Comment by nuevocharrua** [comment]
I'd like to add another observation regarding **buffer_save** and **file_exists**. 
Regardless of the save location, I've found that after using **buffer_save**, using **file_exists** always returns false. 
This raises a question: is the file not being created at all, or are **file_exists** and **buffer_save** using different internal paths?

**Comment by nuevocharrua** [comment]
Just a heads up, I've had to roll back to the previous version of GameMaker to continue working, so I'm unable to create a new project to demonstrate the issue right now. 
The code I've been using is something like this: (save_id is the save slot)
```
// Save a summary in save_id slot
var buffer= buffer_create(1,buffer_grow,1)	
buffer_write(buffer,buffer_text,aux_encode(json_stringify(summary)));
buffer_save(buffer,string("summary_{0}",save_id));
buffer_delete(buffer);
```
```
function get_save_slots()
{
	// list all used slots
	var slots;
	for (var i = 0; i <= 3; ++i) {
	    if file_exists(string("summary_{0}", i)) 
		{
			var buffer = buffer_load(string("summary_{0}", i));
			var json = aux_decode(buffer_read(buffer, buffer_text));
			var summary = json_parse(json);
			summary.exist = true;
			slots[i] = summary;
		}else
		{
			slots[i] = {exist: false, save_id: i}
		}
	}
	
	return slots;
}
```
I should mention that this code works fine in GameMaker 2023.11 on Windows, and also in GameMaker 2023.8 on both Windows and macOS. The problem seems specific to GameMaker 2023.11 on macOS.

**Comment by NanoSharkGames** [comment]
> Please attach a project and I will take a look - it depends on how it is getting used.... they should be saving to the same place (everything routes through the same location) but I want to see how you are using these functions.

[BufferSaveTest.yyz.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13620239/BufferSaveTest.yyz.zip)

Sandbox mode in attached project is not disabled.

**Comment by jmperez127** [comment]
I also updated my GM version today, and I'm facing the same issue.
My project is not in sandbox mode, and the file doesn't seem to be created at all.
I'm using MacOS as well.

**Comment by rwkay** [comment]
Fixed in 2023.11.1
